Library/Homebrew/diagnostic.rb

```diff
@@ -855,6 +855,9 @@ def check_deleted_formula
 
             false
           end
+          if !loadable && tap && !tap.official?
+            loadable = Formulary::FromTapLoader.try_new("#{tap}/#{keg.name}", warn: false).present?
```

Comment:
This seems rather complicated.

Above, we exclude all taps that don't match if `tap` is specified, so I think we can just add `FromTapLoader` to the list and always try loading `"#{tap}/#{keg.name}"` if `tap` is specified, that way we also don't need the additional check above.

Reply:
Please stop dismissing my reviews as “personal preference” when I am using my best judgment to review PRs. It was objectively correct as-is, but it was also objectively not DRY, since two checks were doing the same thing.

> save it for "if merged: this PR will cause a major issue for users"

Yes, I will try to suggest rather than request changes.

Reply:
> Yes, I will try to suggest rather than request changes.

Thanks 🙇🏻